compiler/src/compiler/ast_to_hir.rs

```diff
@@ -542,6 +464,215 @@ impl<'c> Compiler<'c> {
+                        (
+                            self.push(None, Expression::Int(index as u64), Some("key".to_string())),
+                            self.push(
+                                None,
+                                Expression::Text(it.to_owned()),
+                                Some("rawPath".to_string()),
+                            ),
+                        )
+                    })
+                    .collect();
+                self.push(
+                    None,
+                    Expression::Struct(current_path_content),
+                    Some("currentPath".to_string()),
+                )
+            }
+            Input::ExternalFile(_) => {
+                let message_id = self.push(
+                    None,
+                    Expression::Text(
+                        "File doesn't belong to the currently opened project.".to_string(),
+                    ),
+                    Some("message".to_string()),
+                );
+                self.push(
+                    None,
+                    Expression::Call {
+                        function: panic_id,
+                        arguments: vec![message_id],
+                    },
+                    Some("panicked".to_string()),
+                )
+            }
+            Input::Untitled(_) => {
+                let message_id = self.push(
+                    None,
+                    Expression::Text("Untitled files can't call `useAsset`.".to_string()),
+                    Some("message".to_string()),
+                );
+                self.push(
+                    None,
+                    Expression::Call {
+                        function: panic_id,
+                        arguments: vec![message_id],
+                    },
+                    Some("panicked".to_string()),
+                )
+            }
+        }
+    }
+
+    fn generate_use_asset(&mut self) {
+        // HirId(~:test.candy:useAsset) = lambda { HirId(~:test.candy:useAsset:target) ->
+        //   HirId(~:test.candy:useAsset:panic) = builtinPanic
+        //   HirId(~:test.candy:useAsset:useAsset) = builtinUseAsset
+        //   HirId(~:test.candy:useAsset:key) = int 0
+        //   HirId(~:test.candy:useAsset:raw_path) = text "test.candy"
+        //   HirId(~:test.candy:useAsset:currentPath) = struct [
+        //     HirId(~:test.candy:useAsset:key): HirId(~:test.candy:useAsset:raw_path),
+        //   ]
+        //   HirId(~:test.candy:useAsset:importedFileContent) = call HirId(~:test.candy:useAsset:useAsset) with these arguments:
+        //     HirId(~:test.candy:useAsset:currentPath)
+        //     HirId(~:test.candy:useAsset:target)
+        // }
+
+        let reset_state = self.start_scope();
+        self.prefix_keys.push("useAsset".to_string());
+
+        let lambda_parameter_id = hir::Id::new(
+            self.input.clone(),
+            add_keys(&self.prefix_keys[..], "target".to_string()),
+        );
```

Comment:
Move the empty line to unify this and `generate_use(…)`

```suggestion
        let lambda_parameter_id = hir::Id::new(
            self.input.clone(),
            add_keys(&self.prefix_keys[..], "target".to_string()),
        );

```